pyeparse/ia.py

```diff
@@ -0,0 +1,157 @@
+                                and int(ia_top) < int(yp) < int(ia_bottom):
+                                ias_[jj][ii] = meas
+            for jj in range(n_trials):
+                for ii in range(n_ias):
+                    if isinstance(ias_[jj][ii], list):
+                        ias_[jj][ii] = np.rec.fromarrays(np.ones(n_meas) *
+                                                         np.nan, names=labels)
+        elif depmeas == 'sac':
+            depmeas = raw.discrete['saccades']
+            labels = ['eye', 'stime', 'etime', 'sxp', 'syp',
+                      'exp', 'eyp', 'pv']
+            n_meas = len(labels)
+            for meas in depmeas:
+                eye, stime, etime, sxp, syp, exp, eyp, pv = meas
+                meas = np.array(meas)
+                for jj, trial in enumerate(trials):
+                    tstart, tend = trial
+                    if tstart < stime * 1000 < tend:
+                        fix_order[jj].append(int(stime * 1000))                                
+                        # RECTANGLE id left top right bottom [label]
+                        for ii, ia in enumerate(ias):
+                            _, _, ia_left, ia_top, ia_right, ia_bottom, _ = ia
+                            if int(ia_left) < int(sxp) < int(ia_right) \
+                                and int(ia_top) < int(syp) < int(ia_bottom):
+                                ias_[jj][ii] = meas
+            for jj in range(n_trials):
+                for ii in range(n_ias):
+                    if isinstance(ias_[jj][ii], list):
+                        ias_[jj][ii] = np.rec.fromarrays(np.ones(n_meas) *
+                                                         np.nan, names=labels)
+
+        ia_labels = list(ias[:, -1])
+        # ordered dict
+        temp = [IA(zip(ia_labels, trial)) for trial in ias_]
+        self._data = temp
+
+    def __repr__(self):
+        return '<IA | {0} Trials x {1} IAs>'.format(self.n_trials, self.n_ias)
+
+    def __len__(self):
+        return len(self._data)
+
+    @property
+    def shape(self):
+        return (self.n_trials, self.n_ias)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, string_types):
+            if not idx in self._data[0]:
+                raise KeyError("'%s' is not found." % idx)
+            return np.hstack([datum[idx] for datum in self._data])
+        elif isinstance(idx, int):
+            return self._data[idx]
+        elif isinstance(idx, slice):
+            inst = copy(self)
+            inst._data = self._data[idx]
+            inst.n_trials = len(inst._data)
+            return inst
+        else:
+            raise TypeError('index must be an int, string, or slice')
+
+
+class IA(OrderedDict):
+    def __init__(self, entry):
+        super(IA, self).__init__(entry)
+
+    def __repr__(self):
+        return '<Interest Areas: %s>' % ', '.join(self.keys())
+
+
+def read_ia(filename):
+    ia = open(filename).readlines()
```

Comment:
`with open():` will ensure the file gets closed when you are done with it. On Py3k you get `ResourceWarning`s in some cases if you don't do it.